Std/Tactic/Basic.lean

```diff
@@ -162,3 +162,19 @@ example (P : (Nat → Nat) → Prop) : P (fun n => n - n) := by
 -/
 macro (name := Conv.equals) "equals " t:term " => " tac:tacticSeq : conv =>
   `(conv| tactic => show (_ = $t); next => $tac)
+
+/-- `lemma` is not supported, please use `theorem` or import `Mathlib.Tactic.Basic` -/
+syntax (name := lemma) declModifiers
+  group("lemma " declId ppIndent(declSig) declVal Parser.Command.terminationSuffix) : command
+
+/-- `lemma` is not supported, please use `theorem` or import `Mathlib.Tactic.Basic` -/
+@[command_elab «lemma»] def elabLemma : CommandElab := fun stx => do
```

Comment:
Should we do something like
```suggestion
@[command_elab «lemma»] def elabLemma (allow : Bool := false) : CommandElab := fun stx => do
```
so that in mathlib we can just call the same elaborator, with the warning code path skipped?

Reply:
I made an attempt at implementing this.